Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
         #region Assignment solutions
         public static void Assignment1(int low, int high, out int res)
         {
-            /*
-             * TODO: Write code here
-             */
+            if (high < low) // input is invalid
+            {
+                res = -1;
+            }
+            else
+            {
+                Random rnd = new Random();
+                res = rnd.Next(low, high);
+            }
         }
 
         public static void Assignment3()
-        {
-            /*
-             * TODO: Write code here
-             */
+        {            
+            var strToMove = new Dictionary<string, RockPaperScissors.RPS>()
+            {
+                {"Rock", RockPaperScissors.RPS.Rock},
+                {"Paper", RockPaperScissors.RPS.Paper},
+                {"Scissors", RockPaperScissors.RPS.Scissors},
+            };
+            
+            try
+            {
+                Console.WriteLine("Enter player 1 move.");
+                RockPaperScissors.RPS move1 = strToMove[Console.ReadLine()];
+                Console.WriteLine("Enter player 2 move.");
+                RockPaperScissors.RPS move2 = strToMove[Console.ReadLine()];
+
+                if (move1 == move2)
+                    Console.WriteLine("Tie.");
+                else if ((move1 == RockPaperScissors.RPS.Rock && move2 == RockPaperScissors.RPS.Paper)
+                        || move1 == RockPaperScissors.RPS.Paper && move2 == RockPaperScissors.RPS.Scissors
+                        || move1 == RockPaperScissors.RPS.Scissors && move2 == RockPaperScissors.RPS.Rock
+                        )
+                    Console.WriteLine("Player 2 wins.");
+                else
+                    Console.WriteLine("Player 1 wins.");
+            }
+            catch (System.Collections.Generic.KeyNotFoundException)
```

Comment:
You can check the input before making such bold indexing.

Reply:
Right, added move input function in a helper.